pinot-query-planner/src/main/java/org/apache/calcite/rel/rules/PinotAggregateExchangeNodeInsertRule.java

```diff
@@ -83,8 +83,8 @@ public boolean matches(RelOptRuleCall call) {
     }
     if (call.rel(0) instanceof Aggregate) {
       Aggregate agg = call.rel(0);
-      return !agg.getHints().contains(PinotRelationalHints.AGG_LEAF_STAGE)
-          && !agg.getHints().contains(PinotRelationalHints.AGG_INTERMEDIATE_STAGE);
+      return !agg.getHints().contains(AggregateNode.INTERMEDIATE_STAGE_HINT)
```

Comment:
not really. as i mentioned above these 2 hints are not suppose to be directly used by user. this PR only changes the relhint to sqlhint. we are actually not planning to open these 2 up to users. 

do you prefer me changing this to a more explicit internal wording?

Reply:
for now we will add a safety check in AggregateNode to not allow both hints at the same time.

Reply:
> these 2 hints are not suppose to be directly used by user. 

You both may have already discussed this but not sure I follow this. At least in the other PR, we want to create a hint and expose it to user. 

Fair to say that if I want to add a new hint and expose it to the user, this will have to be evolved / extended or there is a different plan ?

```
public static final HintStrategyTable PINOT_HINT_STRATEGY_TABLE = HintStrategyTable.builder()
      .hintStrategy(INTERNAL_AGG_INTERMEDIATE_STAGE, HintPredicates.AGGREGATE)
      .hintStrategy(INTERNAL_AGG_FINAL_STAGE, HintPredicates.AGGREGATE)
```